purchase_merge/wizard/purchase_merge.py

```diff
@@ -0,0 +1,235 @@
+
+    # ----------------------------------------
+    # Update method
+    # ----------------------------------------
+
+    @api.model
+    def _update_values(self, src_purchase, dst_purchase):
+        """Update values of dst_purchase with the ones from the src_purchase.
+        :param src_purchase : recordset of source purchase.order
+        :param dst_purchase : record of destination purchase.order
+        """
+        # merge all order lines + set origin and partner_ref
+        dst_purchase.write(self._get_update_values(src_purchase, dst_purchase))
+        for po in src_purchase:
+            self._add_message("to", [dst_purchase.name], po)
+
+        po_names = src_purchase.mapped("name")
+        self._add_message("from", po_names, dst_purchase)
+
+    @api.model
+    def _get_update_values(self, src_purchase, dst_purchase):
+        """Generate values of dst_purchase with the ones from the src_purchase.
+        :param src_purchase : recordset of source purchase.order
+        :param dst_purchase : record of destination purchase.order
+        """
+        # initialize destination origin and partner_ref
+        origin = {dst_purchase.origin or ""}
+        origin.update({x.origin for x in src_purchase if x.origin})
+
+        partner_ref = {dst_purchase.partner_ref or ""}
+        partner_ref.update({x.partner_ref for x in src_purchase if x.partner_ref})
+
+        # Generate destination origin and partner_ref
+        src_order_line = src_purchase.mapped("order_line")
+
+        # Copy order lines without triggering compute methods
+        order_lines = [(4, line.id, 0) for line in src_order_line]
+
+        return {
+            "order_line": order_lines,
+            "origin": ", ".join(origin),
+            "partner_ref": ", ".join(partner_ref),
+        }
+
+    def _add_message(self, way, po_name, po):
+        """Send a message post with to advise the po about the merge.
+        :param way : choice between 'from' or 'to'
+        :param po_name : list of purchase order name to add in the body
+        :param po_name : the po where the message will be posted
+        """
+        subject = "Merge purchase order"
+        body = _(
+            "This purchase order lines have been merged %(way)s : %(po_names)s",
+            way=way,
+            po_names=" ,".join(po_name),
+        )
+
+        po.message_post(body=body, subject=subject)
+
+    def _merge(self, purchases, dst_purchase=None):
+        """private implementation of merge purchase
+        :param purchases : ids of purchase to merge
+        :param dst_purchase : record of destination purchase.order
+        """
+        if len(purchases) < 2:
+            return
+        record_ids = purchases - dst_purchase
+        openupgrade_merge_records.merge_records(
+            env=self.env,
+            model_name=self._name,
+            record_ids=record_ids.ids,
+            target_record_id=dst_purchase.id,
+        )
```

Comment:
```suggestion
        openupgrade_merge_records.merge_records(
            env=self.env,
            model_name="purchase.order",
            record_ids=record_ids.ids,
            target_record_id=dst_purchase.id,
            delete=False,
        )
```

I tested this module and found that this `openupgrade_merge_records.merge_records` does not work as expected. The field values from the source records are not applied to the target record, for example, the payment term.

The point was mentioned here https://github.com/OCA/purchase-workflow/pull/1679#discussion_r2156131208

@quoc-pn I think we should submit a new PR for this issue so that this PR can focus on migration.

Reply:
@cuongnmtm Thanks for the review, that was an issue I hadn’t noticed.
I think the code fix can go in this migration PR — I’ve reviewed and fixed it.
I’ve also added a test to verify it works correctly.
Please try it again and let me know what you think.

cc: @quoc-pn @Reyes4711-S73